libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,776 @@
+
+    for( index = 0U; index < INVALID_HOSTNAME_LENGTH; index++ )
+    {
+        hostNameBuffer[ index ] = '1';
+    }
+
+    invalidServerInfo.hostNameLength = strlen( hostNameBuffer );
+
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &invalidServerInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Connect fails on socket creation,
+ *  and verify the error code.
+ */
+void test_SecureSocketsTransport_Connect_Insufficient_Memory( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Socket_ExpectAnyArgsAndReturn( SOCKETS_INVALID_SOCKET );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &serverInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Connect fails, when setting
+ * the socket options to require TLS, and verify the error code.
+ */
+void test_SecureSocketsTransport_Connect_Invalid_Credentials_SetRequireTLS( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Socket_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_SetSockOpt_ExpectAnyArgsAndReturn( MOCK_SECURE_SOCKET_ERROR );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &serverInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Connect fails to set the ALPN
+ * protocols, and verify the error code.
+ */
+void test_SecureSocketsTransport_Connect_Invalid_Credentials_AlpnProtos( void )
+{
+    TransportSocketStatus_t returnStatus;
+    SocketsConfig_t localSocketsConfig =
+    {
+        .enableTls         = true,
+        .pAlpnProtos       = ALPN_PROTOS,
+        .maxFragmentLength = MFLN,
+        .disableSni        = true,
+        .pRootCa           = NULL,
+        .rootCaSize        = 0,
+        .sendTimeoutMs     = TEST_TRANSPORT_SND_TIMEOUT_MS,
+        .recvTimeoutMs     = TEST_TRANSPORT_RCV_TIMEOUT_MS
+    };
+
+    SOCKETS_Socket_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_SetSockOpt_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
```

Comment:
Please also update these expectations to test for the arguments passed by the function under test. If an argument (like a pointer) cannot be easily compared against, then the `IgnoreArg` function can be called.
Here is the CMock guide for reference: https://github.com/ThrowTheSwitch/CMock/blob/master/docs/CMock_Summary.md

```suggestion
    SOCKETS_Socket_ExpectAndReturn(SOCKETS_AF_INET, SOCKETS_SOCK_STREAM, SOCKETS_IPPROTO_TCP, 1 /* socket value */);
    SOCKETS_SetSockOpt_ExpectReturn( 1 /* socket value */, 0, SOCKETS_SO_ALPN_PROTOCOLS, NULL, 0,  SOCKETS_ERROR_NONE  );
   SOCKETS_SetSockOpt_IgnoreArg_pvOptionValue();
   SOCKETS_SetSockOpt_IgnoreArg_xOptionLength();
```